File: tareas_go/commands/eliminar.go
package commands

import (
	"errors"
	"fmt"
	"os"
	"tareas_go/estructuras"
)

func (g *GestorTareas) EliminarTarea(id int) error {
	for i, t := range g.tareas {
		if t.ID == id {
			g.tareas = append(g.tareas[:i], g.tareas[i+1:]...)
			g.GuardarTareasJSON()
			return nil
		}
	}
	return errors.New("no se encontró la tarea para eliminar")
}

func (g *GestorTareas) EliminarTodasLasTareas() error {

	if len(g.tareas) == 0 {
		return errors.New("no hay tareas para eliminar")
	}

	fmt.Println("Eliminando todas las tareas...")

	g.tareas = []estructuras.Tarea{}

	err := os.Remove(archivoTareas)
	if err != nil && !os.IsNotExist(err) {
		return fmt.Errorf("error al eliminar archivo JSON: %v", err)
	}

	g.GuardarTareasJSON()
	g.nextID = 1

	fmt.Println("Todas las tareas han sido eliminadas correctamente.")
	return nil
}

File: tareas_rust/src/commands/eliminar.rs
use crate::commands::persistencia;
use crate::commands::gestor::GestorTareas;

pub fn eliminar_tarea(g: &mut GestorTareas, id: i32) -> Result<(), String> {
    if let Some(pos) = g.tareas.iter().position(|t| t.id == id) {
        g.tareas.remove(pos);
        persistencia::guardar_tareas_json(&g.tareas)?;
        println!("Tarea {} eliminada correctamente.", id);
        return Ok(());
    }
    Err(format!("No se encontró la tarea con id {}.", id))
}

pub fn eliminar_todas_las_tareas(g: &mut GestorTareas) -> Result<(), String> {
    if g.tareas.is_empty() {
        return Err("No hay tareas para eliminar.".to_string());
    }
    g.tareas.clear();
    
    match std::fs::remove_file("tareas.json") {
        Ok(_) => (),
        Err(e) if e.kind() == std::io::ErrorKind::NotFound => (),
        Err(e) => return Err(format!("Error al eliminar archivo JSON: {}", e)),
    }

    persistencia::guardar_tareas_json(&g.tareas)?;
    g.next_id = 1;
    println!("Todas las tareas han sido eliminadas correctamente.");
    Ok(())
}
